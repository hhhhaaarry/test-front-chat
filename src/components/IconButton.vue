<template>
   
    <button class="btn btn--icon bg-slate-50 dark:bg-gray-800 p-2 flex justify-center items-center hover:border-blue-500 dark:hover:border-gray-200 ">
        <slot/>
    </button>
</template>

<script setup>

</script>

<style lang="scss" scoped>

</style>