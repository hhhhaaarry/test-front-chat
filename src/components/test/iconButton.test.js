import { mount } from '@vue/test-utils'
import IconButton from '../IconButton.vue'

describe('IconButton.vue', () => {
  // Este test verifica si el componente se monta correctamente
  it('se monta correctamente', () => {
    const wrapper = mount(IconButton)
    expect(wrapper.html()).toMatchSnapshot();
  })

  // Este test verifica si el componente emite un evento cuando se hace clic en Ã©l
  it('emite un evento al hacer clic', async () => {
    const wrapper = mount(IconButton)
    await wrapper.trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
  })
})