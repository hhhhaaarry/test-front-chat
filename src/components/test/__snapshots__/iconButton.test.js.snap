// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconButton.vue > se monta correctamente 1`] = `"<button class=\\"btn btn--icon bg-slate-50 dark:bg-gray-800 p-2 flex justify-center items-center hover:border-blue-500 dark:hover:border-gray-200\\"></button>"`;
